File: apps/wow-minmax-ui/src/app/modules/statistics/services/stats.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';
import { AbilityStats } from '../model/AbilityStats';
import { CharacterStats } from '../model/CharacterStats';
import { RotationStats } from '../model/RotationStats';
import { SpecialAbilityStats } from '../model/SpecialAbilityStats';
import { TalentStats } from '../model/TalentStats';

@Injectable({
	providedIn: 'root'
})
export class StatsService {
	private readonly apiUrl = environment.statsApiUrl;

	constructor(private http: HttpClient) { }

	getAbilityStats(playerId: string) {
		return this.http.get<AbilityStats[]>(`${this.apiUrl}/${playerId}/ability`);
	}

	getCharacterStats(playerId: string) {
		return this.http.get<CharacterStats[]>(`${this.apiUrl}/${playerId}/character`);
	}

	getSpecialAbilities(playerId: string) {
		return this.http.get<SpecialAbilityStats[]>(`${this.apiUrl}/${playerId}/special`);
	}

	getRotationStats(playerId: string) {
		return this.http.get<RotationStats>(`${this.apiUrl}/${playerId}/rotation`);
	}

	getTalentStats(playerId: string) {
		return this.http.get<TalentStats[]>(`${this.apiUrl}/${playerId}/talent`);
	}
}

File: services/wow-estimator/src/main/java/wow/estimator/service/StatsService.java
package wow.estimator.service;

import wow.commons.model.spell.AbilityId;
import wow.estimator.model.*;

import java.util.List;

/**
 * User: POlszewski
 * Date: 2025-03-17
 */
public interface StatsService {
	List<AbilityStats> getAbilityStats(Player player, List<AbilityId> abilityIds, boolean usesCombatRatings, double equivalentAmount);

	CharacterStats getCharacterStats(Player player, boolean worldBuffsAllowed);

	List<SpecialAbilityStats> getSpecialAbilityStats(Player player);

	RotationStats getRotationStats(Player player, Rotation rotation);

	List<TalentStats> getTalentStats(Player player);
}
